src/libasr/ASR.asdl

```diff
@@ -346,6 +350,7 @@ cast_kind
     | IntegerToReal
     | LogicalToReal
     | RealToReal
+    | TemplateToReal
```

Comment:
When is this cast used?

Reply:
It's used in `generics_list_01.py`. The type of `sum` in the division `sum/k` is generic, it may or may not have to be cast into a real number for the division.

Reply:
I see. It's casting to real, but we don't know from "what". Good for now. We can iterate on the design later.